app/routes.py

```diff
@@ -1,2 +1,178 @@
+        return jsonify(video.video_info()), 201
+    return jsonify({"details": "Invalid data"}), 400
+
+@videos_bp.route("/<int:video_id>", methods=["PUT"], strict_slashes=False)
+def update_video(video_id):
+    request_body = request.get_json()
+    video = Video.query.get_or_404(video_id)
+    if "title" in request_body and "release_date" in request_body and "total_inventory" in request_body:
+        video.title = request_body["title"]
+        video.release_date = request_body["release_date"]
+        video.total_inventory = request_body["total_inventory"]
+        video.available_inventory = request_body["total_inventory"]
+        db.session.commit()
+        return jsonify(video.video_info()), 200
+    else:
+        return jsonify({"details": "Invalid data"}), 400
+
+@videos_bp.route("/<video_id>", methods=["DELETE"], strict_slashes=False)
+def delete_video(video_id):
+    vid = Video.query.get_or_404(video_id)
+    db.session.delete(vid)
+    db.session.commit()
+    return make_response({"id": vid.video_id}, 200)
+
+def is_int(value):
+    try:
+        return int(value)
+    except ValueError:
+        return None
+
+@rentals_bp.route("/check-out", methods= ["POST"], strict_slashes=False)
+def rent_video():
+    request_body = request.get_json()
+    if "customer_id" not in request_body or "video_id" not in request_body:
+        return make_response({"details": "Invalid data"}, 400)
+    checkem_out = request_body["customer_id"]
+    check_video = request_body["video_id"]
+    if not is_int(checkem_out) or not is_int(check_video):
+        return make_response({"details": "Bad request"}, 400)
+    customer = Customer.query.get(checkem_out)
+    video = Video.query.get(check_video)
+    if not video.has_available_inventory(): 
+        return make_response({"details": "Inventory not available"}, 400)
+    video.check_out() 
+    customer.videos_checked_out_count += 1
+    this_rental = Rental(**request_body) 
+    db.session.add(this_rental)
+    db.session.commit()
+    return make_response(this_rental.rental_info(), 200)
+    
+@rentals_bp.route("/check-in", methods=["POST"], strict_slashes=False)
+def return_video():
+    request_body = request.get_json()
+    if "customer_id" not in request_body or "video_id" not in request_body:
+        return make_response({"details": "Invalid data"}, 400)
+    checkin_customer = request_body["customer_id"]
+    checkin_video = request_body["video_id"]
+    if not is_int(checkin_customer) or not is_int(checkin_video):
+        return make_response({"details": "Bad request"}, 400)
+    this_rental = Rental.query.filter_by(customer_id = checkin_customer, video_id = checkin_video).first()
+    customer = Customer.query.get(checkin_customer)
+    video = Video.query.get(checkin_video)
+    this_customer = Customer.query.get(checkin_customer)
+    if this_customer == None:
+        return make_response({"details": "Invalid Customer"}, 400)
+
+    if this_customer.videos_checked_out_count > 0:
```

Comment:
This check technically still doesn't cover the case where a customer has a video checked out but it isn't the video that has been requested.
This case would need to be handled by either deleting the rental from the Rentals table on check-in, or adding a column to the Rentals table to note when a Rental has been checked in. 
Happy to talk more about that in a 1:1 if you have any questions!